-1

Tested at: 0572ef0ff30ae4ab296fe7d2aacc85086bcb29ab

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-17469/17711/summary.html

I found follow errors while testing this PR

**Failed tests:  Build  ClangBuild**

* **Build**:

I found an error when building: 
 
 <pre>>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2017-02-08-2300/src/CaloOnlineTools/HcalOnlineDb/src/DBlmapWriter.cc 
>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2017-02-08-2300/src/CaloOnlineTools/HcalOnlineDb/src/HcalChannelIterator.cc 
>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2017-02-08-2300/src/CaloOnlineTools/HcalOnlineDb/src/HcalDbOmds.cc 
>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2017-02-08-2300/src/CaloOnlineTools/HcalOnlineDb/src/WriteL1TriggerObjetsXml.cc 
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2017-02-08-2300/src/CalibCalorimetry/CaloTPG/src/CaloTPGTranscoderULUT.cc: In member function 'unsigned int CaloTPGTranscoderULUT::getOutputLUTSize(const HcalTrigTowerDetId&) const':
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2017-02-08-2300/src/CalibCalorimetry/CaloTPG/src/CaloTPGTranscoderULUT.cc:207:10: error: expected primary-expression before 'else'
          else if (id.ietaAbs() <= theTopology->lastHERing())
          ^
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2017-02-08-2300/src/CalibCalorimetry/CaloTPG/src/CaloTPGTranscoderULUT.cc:209:10: error: 'else' without a previous 'if'
          else
          ^
</pre>

* **Clang**:

I found a compilation error while trying to compile with clang: 
I used this command: 
 scram b vclean && scram build -k -j 48 USER_CXXFLAGS='-fsyntax-only' COMPILER='llvm compile'


 <pre>>> Compiling edm plugin /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2017-02-08-2300/src/Geometry/HcalTowerAlgo/test/HcalGeometryDump.cc 
>> Compiling edm plugin /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2017-02-08-2300/src/Geometry/HcalTowerAlgo/test/HcalGeometryTester.cc 
Entering library rule at src/CalibCalorimetry/CaloTPG/plugins
>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2017-02-08-2300/src/CalibCalorimetry/CaloTPG/src/CaloTPGTranscoderULUT.cc 
>> Compiling edm plugin /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2017-02-08-2300/src/CalibCalorimetry/CaloTPG/plugins/CaloTPGTranscoderULUTs.cc 
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2017-02-08-2300/src/CalibCalorimetry/CaloTPG/src/CaloTPGTranscoderULUT.cc:207:10: error: expected expression
         else if (id.ietaAbs() <= theTopology->lastHERing())
         ^
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2017-02-08-2300/src/CalibCalorimetry/CaloTPG/src/CaloTPGTranscoderULUT.cc:209:10: error: expected expression
         else
         ^
</pre>

--- a/CalibCalorimetry/CaloTPG/src/CaloTPGTranscoderULUT.cc
+++ b/CalibCalorimetry/CaloTPG/src/CaloTPGTranscoderULUT.cc
@@ -202,8 +202,12 @@ CaloTPGTranscoderULUT::getOutputLUTSize(const HcalTrigTowerDetId& id) const
    switch (theTopology->triggerMode()) {
       case HcalTopologyMode::TriggerMode_2009:
       case HcalTopologyMode::TriggerMode_2016:
-      case HcalTopologyMode::TriggerMode_2017:
          return QIE8_OUTPUT_LUT_SIZE;
+      case HcalTopologyMode::TriggerMode_2017:
+         else if (id.ietaAbs() <= theTopology->lastHERing())
+            return QIE8_OUTPUT_LUT_SIZE;
+         else
+            return QIE10_OUTPUT_LUT_SIZE;
       case HcalTopologyMode::TriggerMode_2017plan1:
       case HcalTopologyMode::TriggerMode_2018:
          if (id.ietaAbs() <= theTopology->lastHBRing())

--- a/CalibCalorimetry/CaloTPG/src/CaloTPGTranscoderULUT.cc
+++ b/CalibCalorimetry/CaloTPG/src/CaloTPGTranscoderULUT.cc
@@ -202,11 +202,18 @@ CaloTPGTranscoderULUT::getOutputLUTSize(const HcalTrigTowerDetId& id) const
    switch (theTopology->triggerMode()) {
       case HcalTopologyMode::TriggerMode_2009:
       case HcalTopologyMode::TriggerMode_2016:
-         return QIE8_OUTPUT_LUT_SIZE;
       case HcalTopologyMode::TriggerMode_2017:
+         return QIE8_OUTPUT_LUT_SIZE;
+      case HcalTopologyMode::TriggerMode_2017plan1:
+      case HcalTopologyMode::TriggerMode_2018:
          if (id.ietaAbs() <= theTopology->lastHBRing())
             return QIE8_OUTPUT_LUT_SIZE;
          else if (id.ietaAbs() <= theTopology->lastHERing())
+            return QIE11_OUTPUT_LUT_SIZE;
+         else
+            return QIE10_OUTPUT_LUT_SIZE;
+      case HcalTopologyMode::TriggerMode_2019:
+         if (id.ietaAbs() <= theTopology->lastHERing())
             return QIE11_OUTPUT_LUT_SIZE;
          else
             return QIE10_OUTPUT_LUT_SIZE;

--- a/Geometry/CaloTopology/src/HcalTopology.cc
+++ b/Geometry/CaloTopology/src/HcalTopology.cc
@@ -202,9 +202,9 @@ bool HcalTopology::validHT(const HcalTrigTowerDetId& id) const {
   if (id.iphi()<1 || id.iphi()>IPHI_MAX || id.ieta()==0)  return false;
   if (id.depth() != 0)                              return false;
   if (id.version()==0) {
-    if ((triggerMode_==HcalTopologyMode::TriggerMode_2017 && id.ietaAbs()>28) ||
+    if ((triggerMode_>=HcalTopologyMode::TriggerMode_2017 && id.ietaAbs()>28) ||
 	(id.ietaAbs()>32))                          return false;
-    int ietaMax = (triggerMode_==HcalTopologyMode::TriggerMode_2017) ? 29 : 28;
+    int ietaMax = (triggerMode_>=HcalTopologyMode::TriggerMode_2017) ? 29 : 28;
     if (id.ietaAbs()>ietaMax && ((id.iphi()%4)!=1)) return false;
   } else {
     if (triggerMode_==HcalTopologyMode::TriggerMode_2009) return false;

--- a/Geometry/CaloTopology/src/HcalTopology.cc
+++ b/Geometry/CaloTopology/src/HcalTopology.cc
@@ -202,10 +202,11 @@ bool HcalTopology::validHT(const HcalTrigTowerDetId& id) const {
   if (id.iphi()<1 || id.iphi()>IPHI_MAX || id.ieta()==0)  return false;
   if (id.depth() != 0)                              return false;
   if (id.version()==0) {
-    if ((triggerMode_>=HcalTopologyMode::TriggerMode_2017 && id.ietaAbs()>28) ||
-	(id.ietaAbs()>32))                          return false;
-    int ietaMax = (triggerMode_>=HcalTopologyMode::TriggerMode_2017) ? 29 : 28;
-    if (id.ietaAbs()>ietaMax && ((id.iphi()%4)!=1)) return false;
+    if (id.ietaAbs() > 28) {
+       if (triggerMode_ >= HcalTopologyMode::TriggerMode_2017) return false;
+       if ((id.iphi() % 4) != 1)                               return false;
+       if (id.ietaAbs() > 32)                                  return false;
+    }
   } else {
     if (triggerMode_==HcalTopologyMode::TriggerMode_2009) return false;
     if (id.ietaAbs()<30 || id.ietaAbs()>41)         return false;

--- a/Geometry/HcalCommonData/data/PhaseII/HGCal/hcalRecNumbering.xml
+++ b/Geometry/HcalCommonData/data/PhaseII/HGCal/hcalRecNumbering.xml
@@ -34,7 +34,7 @@
     <PartSelector path="//HCal"/>
     <Parameter name="OnlyForHcalRecNumbering" value="HCAL" eval="false"/>
     <Parameter name="TopologyMode" value="HcalTopologyMode::SLHC" eval="false"/>
-    <Parameter name="TriggerMode"  value="HcalTopologyMode::TriggerMode_2017" eval="false"/>
+    <Parameter name="TriggerMode"  value="HcalTopologyMode::TriggerMode_2019" eval="false"/>
   </SpecPar>
 </SpecParSection>
 </DDDefinition>

--- a/Geometry/HcalCommonData/data/PhaseII/HGCal/hcalRecNumberingRebuild.xml
+++ b/Geometry/HcalCommonData/data/PhaseII/HGCal/hcalRecNumberingRebuild.xml
@@ -28,7 +28,7 @@
     <PartSelector path="//HCal"/>
     <Parameter name="OnlyForHcalRecNumbering" value="HCAL" eval="false"/>
     <Parameter name="TopologyMode" value="HcalTopologyMode::SLHC" eval="false"/>
-    <Parameter name="TriggerMode"  value="HcalTopologyMode::TriggerMode_2017" eval="false"/>
+    <Parameter name="TriggerMode"  value="HcalTopologyMode::TriggerMode_2019" eval="false"/>
   </SpecPar>
 </SpecParSection>
 </DDDefinition>

--- a/Geometry/HcalCommonData/data/PhaseII/NoHE/hcalRecNumbering.xml
+++ b/Geometry/HcalCommonData/data/PhaseII/NoHE/hcalRecNumbering.xml
@@ -21,7 +21,7 @@
     <PartSelector path="//HCal"/>
     <Parameter name="OnlyForHcalRecNumbering" value="HCAL" eval="false"/>
     <Parameter name="TopologyMode" value="HcalTopologyMode::SLHC" eval="false"/>
-    <Parameter name="TriggerMode"  value="HcalTopologyMode::TriggerMode_2017" eval="false"/>
+    <Parameter name="TriggerMode"  value="HcalTopologyMode::TriggerMode_2019" eval="false"/>
   </SpecPar>
 </SpecParSection>
 

--- a/Geometry/HcalCommonData/data/PhaseII/hcalRecNumbering.xml
+++ b/Geometry/HcalCommonData/data/PhaseII/hcalRecNumbering.xml
@@ -28,7 +28,7 @@
     <PartSelector path="//HCal"/>
     <Parameter name="OnlyForHcalRecNumbering" value="HCAL" eval="false"/>
     <Parameter name="TopologyMode" value="HcalTopologyMode::SLHC" eval="false"/>
-    <Parameter name="TriggerMode"  value="HcalTopologyMode::TriggerMode_2017" eval="false"/>
+    <Parameter name="TriggerMode"  value="HcalTopologyMode::TriggerMode_2019" eval="false"/>
   </SpecPar>
 </SpecParSection>
 

--- a/Geometry/HcalCommonData/data/PhaseII/hcalRecNumberingEta4.xml
+++ b/Geometry/HcalCommonData/data/PhaseII/hcalRecNumberingEta4.xml
@@ -32,7 +32,7 @@
     <PartSelector path="//HCal"/>
     <Parameter name="OnlyForHcalRecNumbering" value="HCAL" eval="false"/>
     <Parameter name="TopologyMode" value="HcalTopologyMode::SLHC" eval="false"/>
-    <Parameter name="TriggerMode"  value="HcalTopologyMode::TriggerMode_2017" eval="false"/>
+    <Parameter name="TriggerMode"  value="HcalTopologyMode::TriggerMode_2019" eval="false"/>
   </SpecPar>
 </SpecParSection>
 </DDDefinition>

--- a/Geometry/HcalCommonData/data/PhaseII/hcalRecNumberingRebuild.xml
+++ b/Geometry/HcalCommonData/data/PhaseII/hcalRecNumberingRebuild.xml
@@ -36,7 +36,7 @@
     <PartSelector path="//HCal"/>
     <Parameter name="OnlyForHcalRecNumbering" value="HCAL" eval="false"/>
     <Parameter name="TopologyMode" value="HcalTopologyMode::SLHC" eval="false"/>
-    <Parameter name="TriggerMode"  value="HcalTopologyMode::TriggerMode_2017" eval="false"/>
+    <Parameter name="TriggerMode"  value="HcalTopologyMode::TriggerMode_2019" eval="false"/>
   </SpecPar>
 </SpecParSection>
 

--- a/Geometry/HcalCommonData/data/PhaseII/hcalRecNumberingRebuildHE.xml
+++ b/Geometry/HcalCommonData/data/PhaseII/hcalRecNumberingRebuildHE.xml
@@ -37,7 +37,7 @@
     <PartSelector path="//HCal"/>
     <Parameter name="OnlyForHcalRecNumbering" value="HCAL" eval="false"/>
     <Parameter name="TopologyMode" value="HcalTopologyMode::SLHC" eval="false"/>
-    <Parameter name="TriggerMode"  value="HcalTopologyMode::TriggerMode_2017" eval="false"/>
+    <Parameter name="TriggerMode"  value="HcalTopologyMode::TriggerMode_2019" eval="false"/>
   </SpecPar>
 </SpecParSection>
 </DDDefinition>

--- a/Geometry/HcalCommonData/data/Run2/hcalRecNumbering17Plan01.xml
+++ b/Geometry/HcalCommonData/data/Run2/hcalRecNumbering17Plan01.xml
@@ -38,7 +38,7 @@
     <PartSelector path="//HCal"/>
     <Parameter name="OnlyForHcalRecNumbering" value="HCAL" eval="false"/>
     <Parameter name="TopologyMode" value="HcalTopologyMode::SLHC" eval="false"/>
-    <Parameter name="TriggerMode"  value="HcalTopologyMode::TriggerMode_2017" eval="false"/>
+    <Parameter name="TriggerMode"  value="HcalTopologyMode::TriggerMode_2017plan1" eval="false"/>
   </SpecPar>
 </SpecParSection>
 </DDDefinition>

--- a/Geometry/HcalCommonData/interface/HcalTopologyMode.h
+++ b/Geometry/HcalCommonData/interface/HcalTopologyMode.h
@@ -28,8 +28,11 @@ namespace HcalTopologyMode {
 
   enum TriggerMode {
     TriggerMode_2009=0,         // HF is summed in 3x2 regions
-    TriggerMode_2016=1,         // HF is summed in both 3x2 and 1x1 regions 
-    TriggerMode_2017=2          // HF is summed in 1x1 regions
+    TriggerMode_2016=1,         // HF is summed in both 3x2 and 1x1 regions
+    TriggerMode_2017=2,         // HF upgraded to QIE10
+    TriggerMode_2017plan1=3,    // HF upgraded to QIE10, 1 RBX of HE to QIE11
+    TriggerMode_2018=4,         // HF upgraded to QIE10, HE to QIE11
+    TriggerMode_2019=5          // HF upgraded to QIE10, HBHE to QIE11
   };
 }
 

--- a/Geometry/HcalCommonData/src/HcalTopologyMode.cc
+++ b/Geometry/HcalCommonData/src/HcalTopologyMode.cc
@@ -13,4 +13,7 @@ StringToEnumParser<HcalTopologyMode::TriggerMode>::StringToEnumParser() {
   enumMap["HcalTopologyMode::TriggerMode_2009"] = HcalTopologyMode::TriggerMode_2009;
   enumMap["HcalTopologyMode::TriggerMode_2016"] = HcalTopologyMode::TriggerMode_2016;
   enumMap["HcalTopologyMode::TriggerMode_2017"] = HcalTopologyMode::TriggerMode_2017;
+  enumMap["HcalTopologyMode::TriggerMode_2017plan1"] = HcalTopologyMode::TriggerMode_2017plan1;
+  enumMap["HcalTopologyMode::TriggerMode_2018"] = HcalTopologyMode::TriggerMode_2018;
+  enumMap["HcalTopologyMode::TriggerMode_2019"] = HcalTopologyMode::TriggerMode_2019;
 }

--- a/Geometry/HcalTowerAlgo/src/HcalTrigTowerGeometry.cc
+++ b/Geometry/HcalTowerAlgo/src/HcalTrigTowerGeometry.cc
@@ -11,7 +11,7 @@ HcalTrigTowerGeometry::HcalTrigTowerGeometry( const HcalTopology* topology )
    auto tmode = theTopology->triggerMode();
    useRCT_ = tmode <= HcalTopologyMode::TriggerMode_2016;
    use1x1_ = tmode >= HcalTopologyMode::TriggerMode_2016;
-   use2017_ = tmode == HcalTopologyMode::TriggerMode_2017;
+   use2017_ = tmode >= HcalTopologyMode::TriggerMode_2017;
 }
 
 std::vector<HcalTrigTowerDetId> 

